lib/db_ido/dbobject.cpp

```diff
@@ -119,26 +119,17 @@ void DbObject::SendConfigUpdateHeavy(const Dictionary::Ptr& configFields)
 
 	ASSERT(configFields->Contains("config_hash"));
 
-	ConfigObject::Ptr object = GetObject();
+	Dictionary::Ptr configHashField = new Dictionary({{"config_hash", configFields->Get("config_hash")}});
+	configFields->Set("config_hash", Empty);
 
-	DbQuery query;
-	query.Table = GetType()->GetTable() + "s";
-	query.Type = DbQueryInsert | DbQueryUpdate;
-	query.Category = DbCatConfig;
-	query.Fields = configFields;
-	query.Fields->Set(GetType()->GetIDColumn(), object);
-	query.Fields->Set("instance_id", 0); /* DbConnection class fills in real ID */
-	query.Fields->Set("config_type", 1);
-	query.WhereCriteria = new Dictionary({
-		{ GetType()->GetIDColumn(), object }
-	});
-	query.Object = this;
-	query.ConfigUpdate = true;
-	OnQuery(query);
```

Comment:
Wouldn't it be enough to just NULL the config_hash before OnConfigUpdateHeavy() and perform the above query...